### PR TITLE
New setting 'auto_detect_panels'

### PR DESCRIPTION
**Pull request by t-wissmann**
This helps working around other problems with panels, such as the one in
issue #815.

### CODE REVIEW

File: doc/herbstluftwm.txt [hunk 1]
```diff
@@ -1074,6 +1074,11 @@ auto_detect_monitors (Boolean)::
     If set, detect_monitors is automatically executed every time a monitor is
     connected, disconnected or resized.
 
+auto_detect_monitors (Boolean)::
```
Comment (blarz) [inline doc/herbstluftwm.txt:1077]: _panels?

Comment (t-wissmann) [reply doc/herbstluftwm.txt]: Thanks!

**Comment by blarz** [comment]
Doesn't seem to break anything for me, with the default panel and the default setting.
Any other ways to test this?

**Comment by t-wissmann** [comment]
> Doesn't seem to break anything for me, with the default panel and the default setting.
> Any other ways to test this?

We don't have test cases for the panel detection at all (we should have some in the long run). I also tested it locally with `lxpanel` within a Xephyr session, and it seems to be OK.

**Pull request merged**